biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/ABITrace.java

```diff
@@ -0,0 +1,568 @@
+import org.biojava.nbio.core.sequence.compound.DNACompoundSet;
+import org.biojava.nbio.core.sequence.compound.NucleotideCompound;
+import org.biojava.nbio.core.sequence.template.AbstractSequence;
+import org.biojava.nbio.core.exceptions.CompoundNotFoundException;
+
+/**
+ * Title: ABITrace<br><br>
+ * ABITrace is a class for managing ABI file information,
+ * it is capable of opening an ABI file and storing
+ * the most important fields, which can be recalled as simple java types. It can also return
+ * an image corresponding to the trace.
+ * It has three constructors with input types <code>File, URL, and byte[]</code>.<br><br>
+ * ABI files contain two sets of basecall and sequence data, one that was originally
+ * created programatically and the other, which is an editable copy. This version of this object
+ * only references the original unedited data.<br>
+ */
+public class ABITrace {
+
+    //the next three lines are the important persistent data
+    private String sequence;
+    private int A[], G[], C[], T[], Basecalls[], Qcalls[];
+    private int TraceLength, SeqLength;
+
+    //This is the actual file data.
+    private byte[] TraceData;
+
+    private int maximum = 0;
+
+    //the next four declaration lines comprise the file index information
+    private int MacJunk = 0; //sometimes when macintosh files are
+    //FTPed in binary form, they have 128 bytes
+    //of crap pre-pended to them. This constant
+    //allows ABITrace to handle that in a way that
+    //is invisible to the user.
+    private static int AbsIndexBase = 26; //The file location of the Index pointer
+    private int IndexBase, PLOC, PCON;
+
+    //the next declaration is for the actual file pointers
+    private int DATA9, DATA10, DATA11, DATA12, PBAS2, FWO;
+
+    /**
+     * The File constructor opens a local ABI file and parses the content.
+     *
+     * @param ABIFile is a <code>java.io.File</code> on the local file system.
+     * @throws IOException              if there is a problem reading the file.
+     * @throws IllegalArgumentException if the file is not a valid ABI file.
+     */
+    public ABITrace(File ABIFile) throws IOException {
+        byte[] bytes = null;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        FileInputStream fis = new FileInputStream(ABIFile);
+        BufferedInputStream bis = new BufferedInputStream(fis);
+        int b;
+        while ((b = bis.read()) >= 0) {
+            baos.write(b);
+        }
+        bis.close(); fis.close(); baos.close();
+        bytes = baos.toByteArray();
+        initData(bytes);
+    }
+
+    /**
+     * The URL constructor opens an ABI file from any URL.
+     * @param ABIFile is a <code>java.net.URL</code> for an ABI trace file.
+     * @throws IOException if there is a problem reading from the URL.
+     * @throws IllegalArgumentException if the URL does not contain a valid ABI file.
+     */
+    public ABITrace( URL ABIFile ) throws IOException
```

Comment:
This and the method above are very similar. It'd be nice to extract the common part to a separate method to be shared by both.